protected/humhub/modules/admin/controllers/InformationController.php

```diff
@@ -93,17 +95,25 @@ public function actionDatabase(int $migrate = self::DB_ACTION_CHECK)
 
         if ($migrate === self::DB_ACTION_RUN) {
             $migrationService->migrateUp();
-            $migrationOutput = sprintf(
+            Yii::$app->session->set(self::DB_MIGRATION_RESULT_KEY, sprintf(
                 "%s\n%s",
                 $migrationService->getLastMigrationOutput(),
                 SettingController::flushCache()
-            );
-        } else {
+            ));
+            return $this->redirect(['/admin/information/database']);
+        }
+
+        $migrationOutput = Yii::$app->session->get(self::DB_MIGRATION_RESULT_KEY);
+
+        if ($migrationOutput === null) {
```

Comment:
yes, the method `Yii::$app->session->has()` exists, but even if we call it then we must call `Yii::$app->session->get()` in order to have `$migrationOutput` because then it is passed into the view file.
So new code may be like this:

```php
if (Yii::$app->session->has(self::DB_MIGRATION_RESULT_KEY)) {
    $migrationOutput = Yii::$app->session->get(self::DB_MIGRATION_RESULT_KEY);
    Yii::$app->session->remove(self::DB_MIGRATION_RESULT_KEY);
}
```

instead of current:

```php
$migrationOutput = Yii::$app->session->get(self::DB_MIGRATION_RESULT_KEY);
if ($migrationOutput !== null) {
    Yii::$app->session->remove(self::DB_MIGRATION_RESULT_KEY);
}
```

I don't see a reason to call here the `Yii::$app->session->has()` because it does almost same as `Yii::$app->session->get()`.